tools/travis/run_lxc_container.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Start a lxc container to run jobs in travis.
+# Arguments:
+#   name: The name of the container.
+
+set -ev
+
+if [ "$#" -lt 1 ]; then
+    echo "Usage: "$0" <name>"
+    exit 1
+fi
+
+script_path="$(dirname "$0")"
+project_path="$(readlink -f "$script_path/../..")"
+name="$1"
+
+lxc=/snap/bin/lxc
+
+echo "Start the LXC container."
+sudo $lxc launch --ephemeral ubuntu:xenial "$name"
+# This is likely needed to wait for systemd in the container to start and get
+# an IP, configure DNS. First boot is always a bit slow because cloud-init
+# needs to run too.
+sleep 10
```

Comment:
I tried removing the sleep and leaving just the ping, but it wasn't enough. apt update still failed.
I'm trying now with this loop plus a 5s sleep.